### PR TITLE
fix : old published news images not displayed for none mumbers - EXO-64578

### PR DESCRIPTION
**Pull request by sofyenne**
This change will update the permissions for displaying published news images to non-members.

### CODE REVIEW

File: data-upgrade-news/src/main/resources/conf/portal/configuration.xml [hunk 1]
```diff
@@ -254,6 +254,50 @@
...
+    <component-plugin>
+      <name>PublishedNewsImagesPermissionsUpgradePlugin</name>
+      <set-method>addUpgradePlugin</set-method>
+      <type>org.exoplatform.news.upgrade.jcr.PublishedNewsImagesPermissionsUpgradePlugin</type>
+      <description>Update published news images permissions</description>
+      <init-params>
+        <value-param>
+          <name>product.group.id</name>
+          <description>The groupId of the product</description>
+          <value>org.exoplatform.platform</value>
```
Comment (ahamdi) [inline data-upgrade-news/src/main/resources/conf/portal/configuration.xml]:
```suggestion
          <value>org.exoplatform.news</value>
```

File: data-upgrade-news/src/main/resources/conf/portal/configuration.xml [hunk 2]
```diff
@@ -254,6 +254,50 @@
...
+        <value-param>
+          <name>platform.users.group.identity</name>
+          <description>The identity of the platform users group</description>
+          <value>*:/platform/users</value>
+        </value-param>
+        <value-param>
+          <name>permissions</name>
+          <description>Permissions to be added to the published news images</description>
+          <value>read</value>
+        </value-param>
```
Comment (ahamdi) [inline data-upgrade-news/src/main/resources/conf/portal/configuration.xml]: I am not sure if this UP will be reused, is there a need to parameterize those properties ?

File: data-upgrade-news/src/main/resources/conf/portal/configuration.xml [hunk 3]
```diff
@@ -254,6 +254,50 @@
...
+        <value-param>
+          <name>plugin.upgrade.async.execution</name>
+          <description>The plugin will be executed in an asynchronous mode</description>
+          <value>true</value>
+        </value-param>
+        <value-param>
+          <name>plugin.upgrade.target.version</name>
+          <description>Target version of the plugin</description>
+          <value>6.5.x</value>
```
Comment (ahamdi) [inline data-upgrade-news/src/main/resources/conf/portal/configuration.xml]: We should put an exact version here

File: data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  private static final Pattern IMAGE_SRC_PATTERN                = Pattern.compile("src=\"/portal/rest/images/?(.+)?\"");
+  public static final String EXO_PRIVILEGEABLE = "exo:privilegeable";
+  public static final String READ_PERMISSIONS = "permissions";
+  public static final String PLATFORM_USERS_GROUP_IDENTITY = "platform.users.group.identity";
+  private  int imageNewsUpdatedCount;
+  private  int newsCount;
+  private String readPermissions;
+  private String platformUsersGroupIdentity;
+
+  public PublishedNewsImagesPermissionsUpgradePlugin(InitParams initParams, RepositoryService repositoryService, SessionProviderService sessionProviderService) {
```
Comment (ahamdi) [inline data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java]: Could you format this file correctly ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: data-upgrade-news/src/main/resources/conf/portal/configuration.xml (hunk 1 above)
Comment (sofyenne) [reply data-upgrade-news/src/main/resources/conf/portal/configuration.xml]: done !

File: data-upgrade-news/src/main/resources/conf/portal/configuration.xml (hunk 2 above)
Comment (sofyenne) [reply data-upgrade-news/src/main/resources/conf/portal/configuration.xml]: done !

File: data-upgrade-news/src/main/resources/conf/portal/configuration.xml (hunk 3 above)
Comment (sofyenne) [reply data-upgrade-news/src/main/resources/conf/portal/configuration.xml]: done !

File: data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java (hunk 4 above)
Comment (sofyenne) [reply data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java]: done !

**Review by ahamdi — APPROVED** [review]

**Pull request merged**